spring-security/src/test/java/com/sap/cloud/security/spring/autoconfig/XsuaaTokenFlowAutoConfigurationTest.java

```diff
@@ -119,43 +118,24 @@ void autoConfigurationUsesTokenFlowsForMultipleXsuaaServicesAsPrimary() {
 		mt_runner.run(context -> assertTrue(context.containsBean("xsuaaTokenFlows")));
 	}
 
-	@Test
-	void configures_xsuaaMtlsRestTemplateForMultipleXsuaaServicesAsPrimary() {
-		WebApplicationContextRunner mt_runner;
-
-		List<String> mt_properties = new ArrayList<>(properties);
-		mt_properties.add("sap.security.services.xsuaa[0].url:http://localhost");
-		mt_properties.add("sap.security.services.xsuaa[0].clientid:cid");
-		mt_properties.add("sap.security.services.xsuaa[0].certificate:cert");
-		mt_properties.add("sap.security.services.xsuaa[0].key:key");
-		mt_properties.add("sap.security.services.xsuaa[1].clientid:cid");
-
-		mt_runner = new WebApplicationContextRunner()
-				.withPropertyValues(mt_properties.toArray(new String[0]))
-				.withConfiguration(AutoConfigurations.of(HybridIdentityServicesAutoConfiguration.class,
-						XsuaaTokenFlowAutoConfiguration.class));
-
-		mt_runner.run(context -> assertTrue(context.containsBean("xsuaaTokenFlows")));
-		mt_runner.run(context -> assertTrue(context.containsBean("mtlsRestOperations")));
-	}
-
 	@Test
 	void userConfigurationCanOverrideDefaultBeans() {
```

Comment:
I think this test only asserts that the user can override both the HttpClientFactory AND the XsuaaTokenFlows beans in tandem. The more common case which is not tested here, is probably users overriding only the http client.
If the intended way is to override both beans, we should consider adding comments or documentation for the users.
If users should be able to override only the http client bean, we should not use "customHttpClient" as bean qualifier in the test, as this will not be used (because XsuaaTokenFlows bean now has its http client injected with qualifier "tokenFlowHttpClient").

Reply:
good point, but i can't even test it as it has to be the same name for the bean, but yeah readme need to be updated 👍🏻

Reply:
it's actually already described in readme with a link to spring boot documentation, so seems to be all good here